doc/source/deploy-on-yarn.rst

```diff
@@ -0,0 +1,276 @@
+
+Skein Configuration
+-------------------
+
+A Ray job is configured to run as two `Skein services`:
+
+1. The ``ray-head`` service that starts the Ray head node and then runs the
+   application.
+2. The ``ray-worker`` service that starts worker nodes that join the Ray cluster.
+   You can change the number of instances in this configuration or at runtime
+   using ``skein scale`` to scale the cluster up/down.
+
+The specification for each service consists of necessary files and commands that will be run to start the service.
+
+.. code-block:: yaml
+
+    services:
+        ray-head:
+            # There should only be one instance of the head node per cluster.
+            instances: 1
+            resources:
+                # The resources for the head node.
+                vcores: 1
+                memory: 2048
+            files:
+                ...
+            script:
+                ...
+        ray-worker:
+            # There should only be one instance of the head node per cluster.
+            instances: 1
+            resources:
+                # The resources for the head node.
+                vcores: 1
+                memory: 2048
+            files:
+                ...
+            script:
+                ...
+
+Packaging Dependencies
+----------------------
+
+Use the ``files`` option to specify files that will be copied into the YARN container for the application to use. See `the Skein file distribution page <https://jcrist.github.io/skein/distributing-files.html>`_ for more information.
+
+.. code-block:: yaml
+
+    services:
+        ray-head:
+            # There should only be one instance of the head node per cluster.
+            instances: 1
+            resources:
+                # The resources for the head node.
+                vcores: 1
+                memory: 2048
+            files:
+                # ray/doc/yarn/example.py
+                example.py: example.py
+            #     # A packaged python environment using `conda-pack`. Note that Skein
+            #     # doesn't require any specific way of distributing files, but this
+            #     # is a good one for python projects.
+            #     # See https://jcrist.github.io/skein/distributing-files.html
+            #     environment: environment.tar.gz
+
+Ray Setup in YARN
+-----------------
+
+Below is a walkthrough of the bash commands used to start the ``ray-head`` and ``ray-worker`` services. Note that this configuration will launch a new Ray cluster for each application, not reuse the same cluster.
+
+Head node commands
+~~~~~~~~~~~~~~~~~~
+
+Start by activating a pre-existing environment for dependency management.
+
+.. code-block:: bash
+
+    source /home/rayonyarn/miniconda3/bin/activate
+
+Obtain the Skein Application ID which is used when pushing addresses to worker services.
+
+.. code-block:: bash
+
+    APP_ID=$(python -c 'import skein;print(skein.properties.application_id)')
```

Comment:
When running as part of an application, you can use `current` instead of specifying the application ID. No need to infer the application ID here.

Reply:
See https://jcrist.github.io/skein/cli.html for reference.